analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+Do originating libraries cluster together?
+
+```{r fig.width=10}
+all_reference_clusters <- has_cnv_df |>
+  dplyr::filter(
+    normal_reference == "all-immune",
+    immune_reference_cell
+  )
+all_plot <- ggplot(all_reference_clusters) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  labs("All immune reference") +
+  theme(legend.position = "bottom")
+
+
+subset_reference_clusters <- has_cnv_df |>
+  # bring in library_id
+  dplyr::inner_join(
+    infercnv_df |>
+      dplyr::select(barcodes, library_id) |>
+      unique()
+  ) |>
+  dplyr::filter(
+    normal_reference == "subset-immune",
+    subset_reference_cell
+  )
+subset_plot <- ggplot(subset_reference_clusters) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  labs("Subset immune reference") +
+  theme(legend.position = "bottom")
+
+wrap_plots(all_plot, subset_plot, guides = "collect") &
+  theme(legend.position = "bottom")
+```
+
+
+How many CNVs were inferred per cluster?
+We expect a single value here, because with `analysis_mode = "subclusters"` (the `inferCNV` default), CNV parameters are estimated per cluster.
+
+```{r}
+# How many CNV per cell per cluster?
+ggplot(has_cnv_df) +
+  aes(x = subcluster, y = total_cnv_per_cell) +
+  geom_point(size = 2) +
+  facet_wrap(
+    vars(normal_reference),
+    scales = "free_x"
+  ) +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+```
+
+
+## CNV across chromosomes
+
+Now, we'll look at CNVs across chromosomes, specifically considering the `proportion_scaled_cnv_` output from the HMM.
+
+Again, these are the specific CNVs we expect to see in Ewing sarcoma:
+
+```{r}
+validation_df
+```
+
+```{r}
+# First, construct data frame with proportion_scaled_ instead of has_cnv
+cnv_validation_df <- infercnv_df |>
+  dplyr::select(
+    barcodes,
+    library_id,
+    normal_reference,
+    starts_with("proportion_scaled_")
+  ) |>
+  # tidy up
+  tidyr::pivot_longer(
+    starts_with("proportion_scaled_"),
+    names_to = "cnv_type_raw",
+    values_to = "proportion"
+  ) |>
+  # more tidying
+  dplyr::mutate(cnv_type_raw = stringr::str_remove(cnv_type_raw, "^proportion_scaled_")) |>
+  tidyr::separate(cnv_type_raw, sep = "_", into = c("cnv_type", "chr")) |>
+  # remove extra chromosomes and cells not in this library, and then can make chr numeric
+  dplyr::filter(
+    !stringr::str_starts(chr, "chrMT"),
+    !stringr::str_starts(chr, "chrGL"),
+    library_id == params$library_id
+  ) |>
+  dplyr::mutate(
+    chr = stringr::str_remove(chr, "^chr"),
+    chr = as.numeric(chr)
+  )
+```
+
+We'll make density plots of the predicted CNV events across chromosomes, for each of the two normal references:
+
+
+```{r fig.height=8, fig.width=8}
+cnv_validation_df |>
+  dplyr::filter(normal_reference == "all-immune") |>
+  ggplot() +
+  aes(
+    x = proportion,
+    color = cnv_type
+  ) +
+  geom_density() +
```

Comment:
```suggestion
  geom_density(bounds = c(0,1), bw = 0.05) +
```
Something is off with these plots since they aren't showing any gains that I would expect to be there based on the heatmap. This is what I used when plotting this value before so maybe this will help? 
Additionally, I might just look at the proportion scaled regardless of up/down right now and color by normal, non-normal reference, and tumor. I want to see if the tumor cells show higher CNV proportions in the chromosomes we expect to be altered. 

Also note that those CNVs are not present in every patient and most are in less than 50% so no guarantee that we'll see the ones listed in the table. This patient should have chr8 though and it looks like there is some slight amplification in the heatmap.